### ISSUE TITLE
monitoring cannot start when runAsNonRoot is set

### ISSUE THREAD

**Issue by tibers** [rancher/rancher#20884]
**What kind of request is this (question/bug/enhancement/feature request):**

Add users to the monitoring pods, don't run everything as root.

**Steps to reproduce (least amount of steps as possible):**

1) Follow the hardening guide (which needs to be freshened anyway)
2) make sure MustRunAsNonRoot is set
3) Attempt to install monitoring

**Result:**

Monitoring pods die with `Error: container has runAsNonRoot and image will run as root`

**EDIT**: Also the Kubectl pod seems to die when this is set. Why would the kubectl console pod be running as root? Fix that up too. :)

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
- Installation option (single install/HA):

Rancher 2.2.4 / HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
(paste the output here)
```

- Docker version (use `docker version`):

```
(paste the output here)
```

**Comment by tibers** [comment]
This issue is related: https://github.com/rancher/rancher/issues/19439

**Comment by jiaqiluo** [comment]
I met the same error in rancher:v2.3.0-rc3

Steps:
- follow the guide here to use rke to provision a k8s cluster in hardening mode: https://rancher.com/docs/rancher/v2.x/en/security/hardening-2.2/
- install Rancher:v2.3.0-rc3
- In rancher, enable Github Auth
- log in as a Github user as a standard user
- add a custom cluster with PSP enabled and configure both nodes and the edit the cluster as YAML following the guide
- enable the cluster monitoring 

Results:
- errors show under the workloads `exporter-kube-state-cluster-monitoring`, `operator-init-cluster-monitoring`, and `operator-init-monitoring-operator`
```
CreateContainerConfigError: container has runAsNonRoot and image will run as root
```
<img width="1414" alt="Screen Shot 2019-09-16 at 4 00 51 PM" src="https://user-images.githubusercontent.com/6218999/64999034-34ddea80-d89b-11e9-839f-aae8a0ed0a4a.png">

- cannot access to kubectl from UI

<img width="1863" alt="Screen Shot 2019-09-16 at 3 59 49 PM" src="https://user-images.githubusercontent.com/6218999/64998985-095b0000-d89b-11e9-990c-7f8091f568c5.png">

**Comment by deniseschannon** [comment]
@alena1108 since this is a community PR fix, we will need to work on cherry-picking their commit into the relevant branch.

**Comment by deniseschannon** [comment]
Since this issue is being fixed in kontainer-driver-metadata, I don't believe a forwardport issue is needed.

**Comment by alena1108** [comment]
The fix is available in monitoring chart v0.0.5

**Comment by jiaqiluo** [comment]
The bug fix is validated in v2.3-head 8659f45 in the hardening cluster mode.

Steps:
- follow the guide here to use rke to provision a k8s cluster in hardening mode: https://rancher.com/docs/rancher/v2.x/en/security/hardening-2.2/
- install Rancher:v2.3-head
- In rancher, enable Github Auth
- log in as a Github user as a standard user
- add a custom cluster with PSP enabled and configure both nodes and the edit the cluster as YAML following the guide
- enable cluster and project monitoring 

Results:
- the default chart version is 0.0.5
- both the cluster and protect monitoring work as expected.
